File: i2f-extension/i2f-extension-swl/src/main/java/i2f/extension/swl/impl/sm/antherd/supplier/SwlAntherdSm2AsymmetricEncryptorSupplier.java
package i2f.extension.swl.impl.sm.antherd.supplier;

import i2f.extension.swl.impl.sm.antherd.SwlAntherdSm2AsymmetricEncryptor;
import i2f.swl.std.ISwlAsymmetricEncryptor;
import i2f.swl.std.supplier.ISwlAsymmetricEncryptorSupplier;

/**
 * @author Ice2Faith
 * @date 2024/7/10 9:45
 * @desc
 */
public class SwlAntherdSm2AsymmetricEncryptorSupplier implements ISwlAsymmetricEncryptorSupplier {

    @Override
    public ISwlAsymmetricEncryptor get() {
        return new SwlAntherdSm2AsymmetricEncryptor();
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/extension/impl/sm/antherd/supplier/SwlAntherdSm2AsymmetricEncryptorSupplier.js
import SwlAntherdSm2AsymmetricEncryptor from "../SwlAntherdSm2AsymmetricEncryptor";

/**
 * @return {SwlAntherdSm2AsymmetricEncryptorSupplier}
 * @implements ISwlAsymmetricEncryptorSupplier
 */
function SwlAntherdSm2AsymmetricEncryptorSupplier() {

}

/**
 * @return {ISwlAsymmetricEncryptor}
 */
SwlAntherdSm2AsymmetricEncryptorSupplier.prototype.get = function () {
    return new SwlAntherdSm2AsymmetricEncryptor()
}

export default SwlAntherdSm2AsymmetricEncryptorSupplier
